domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsController.kt

```diff
@@ -30,14 +32,17 @@ class PerformanceMetricsController @Inject constructor(
   private val exceptionLogger: ExceptionLogger,
   private val performanceMetricsEventLogger: PerformanceMetricsEventLogger,
   cacheStoreFactory: PersistentCacheStore.Factory,
-  @PerformanceMetricsLogStorageCacheSize private val metricLogStorageCacheSize: Int
+  @PerformanceMetricsLogStorageCacheSize private var metricLogStorageCacheSize: Int
 ) {
 
   private var isAppInForeground: Boolean = false
 
   private val metricLogStore =
     cacheStoreFactory.create("metric_logs", OppiaMetricLogs.getDefaultInstance())
 
+  private val lastCpuUsageParameters =
```

Comment:
Do we actually need to cache this on disk? It's going to bias the results since it's much more expensive to store on disk than in-memory. I'm assuming we can just throw away results if the app is killed by the system.

Reply:
How will this create a bias? The main reason why I think on-disk caching would be beneficial is because in cases like background logging where time interval is of 60 minutes, there'll be very less instances with existing previous values and hence the readings won't be accurate. On-disk caching will only be of last 1 value and not of a list of values so I don't think it'll be that expensive here.

Reply:
Well, operationally it requires significant more CPU time to write something to disk than to ram. It's not a lot of CPU time overall, but when profiling the CPU we generally want to exit the critical sections as quickly as possible to ensure the vast majority of inspection time is spent actually executing stuff we care to profile (i.e. app code).

That being said, will caching actually work? The time won't match up when trying to compute the difference in CPU time, or am I maybe missing something?

Reply:
as discussed in chat after this, I've modified the implementation to use MutableStateFlow instead.